src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkRuleSectionAfterExecutionInspection.kt

```diff
@@ -50,36 +48,36 @@ class SmkRuleSectionAfterExecutionInspection : SnakemakeInspection() {
                                 ),
                                 ProblemHighlightType.ERROR,
                                 null,
-                                SwapTwoSectionsQuickFix(SmartPointerManager.createPointer(executionSection))
+                                MoveExecutionSectionToEndOfRuleQuickFix(SmartPointerManager.createPointer(executionSection))
                         )
                     }
                 }
             }
         }
     }
 
-    private class SwapTwoSectionsQuickFix(private val precedingSectionPointer: SmartPsiElementPointer<SmkSection>) : LocalQuickFix {
-        override fun getFamilyName() = SnakemakeBundle.message("INSP.INTN.move.rule.section.upwards.family")
+    private class MoveExecutionSectionToEndOfRuleQuickFix(
+            private val executionSectionPointer: SmartPsiElementPointer<SmkSection>
+    ) : LocalQuickFix {
+        override fun getFamilyName() = SnakemakeBundle.message("INSP.INTN.move.execution.section.down.family")
 
         override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
-            val firstSection = precedingSectionPointer.element ?: return
-            val secondSection = descriptor.psiElement
+            val executionSection = executionSectionPointer.element ?: return
+            val executionSectionWhitespace = executionSection.prevSibling as? PsiWhiteSpace ?: return
+            val containingRule = PsiTreeUtil.getParentOfType(executionSection, SmkRuleOrCheckpoint::class.java)!!
             val document = FileEditorManager.getInstance(project).selectedTextEditor?.document ?: return
-            val startOffset1 = firstSection.textOffset
-            val endOffset1 = firstSection.textRange.endOffset
-            val initialStartOffset2 = secondSection.textOffset
-            val initialEndOffset2 = secondSection.textRange.endOffset
-            val text1 = firstSection.text
-            val text2 = secondSection.text
+            val containingRuleEndOffset =
+                    containingRule.textRange.startOffset + containingRule.textLength -
+                            (executionSection.textLength + executionSectionWhitespace.textLength)
             WriteCommandAction.runWriteCommandAction(project) {
-                document.replaceString(startOffset1, endOffset1, text2)
-                if (text1.length > text2.length) {
-                    val endOffset = startOffset1 + text2.length
-                    document.replaceString(startOffset1 + text2.length, endOffset, "")
-                }
-                val startOffset2 = initialStartOffset2 + (text2.length - text1.length)
-                val endOffset2 = initialEndOffset2 + (text2.length - text1.length)
-                document.replaceString(startOffset2, endOffset2, text1)
+                document.deleteString(
```

Comment:
Could you implement it using PsiElement add/delete/replace/etc operations? Smth like in `PyRemoveArgumentQuickFix`?

Reply:
done